frontend/src/static/js/v_summary.js

```diff
@@ -379,6 +379,12 @@ window.vSummary = {
       });
     },
 
+    getRandomColor() {
+      const maxHexColorValue = 16777214;
+      // excludes #000000 and #FFFFFF as they are reserved
+      return `#${Math.round(Math.random() * maxHexColorValue + 1).toString(16)}`;
```

Comment:
instead of it being purely random, such that color collision can happen, can we reuse the existing template colors at different shade? Or apply an algorithm to check that the new colors are not similar to any of the existing colors?
https://stackoverflow.com/questions/5392061/algorithm-to-check-similarity-of-colors

Reply:
I modified the implementation such that it will keep generating random colours until a colour that is not similar to the existing colours. The distance used to compare colours is Delta E, which is a perceptual colour distance. The threshold that was selected is 11, which strikes a balance between perceptibility and the time required generate a colour that satisfies the threshold. The threshold is based on the table below, available [here](http://zschuessler.github.io/DeltaE/learn/):

<img width="658" alt="Screen Shot 2020-04-09 at 2 16 29 PM" src="https://user-images.githubusercontent.com/32864116/78863653-b7630d00-7a6c-11ea-98b7-b3f03a74466c.png">

Also, since we can't really reproduce predefined colours being used up in our preview, I created a preview just to show how the generated colours will look like [here](http://generatenonrepeatingcolors-deltae.surge.sh/#search=&sort=groupTitle&sortWithin=title&timeframe=commit&mergegroup=false&groupSelect=groupByRepos&breakdown=true).

Note that the colours will be generated **_once for every page reload_**, so I don't think it's a big overhead to add this algorithm.

Reply:
Sounds good to me. If this works well, we can consider removing the existing `selectedColors` template. 

To make the algorithm more deterministic, we can consider using a const random seed.

Reply:
I have utilised an external [seeded random generator](https://github.com/davidbau/seedrandom) to do this, since javascript does not have seeding capability for `Math.random`. I have also removed the existing color templates. Here is the [updated dashboard link](http://updatedrandomcolorgenerator.surge.sh/#search=&sort=groupTitle&sortWithin=title&timeframe=commit&mergegroup=false&groupSelect=groupByRepos&breakdown=true) that shows the generated file type colors. Reloading the same page will result to the same colors.

@eugenepeh @damithc do let me know if this is ok, or anymore changes required (such as increasing the threshold so that the colors are more different than one another)